examples/external-data/src/mock-customer-service/service.ts

```diff
@@ -12,29 +12,64 @@ import fetch from "node-fetch";
 import { ClientManager } from "../utilities";
 import { assertValidTaskData, ITaskData } from "../model-interface";
 
+/**
+ * Expected shape of the "broadcast-signal" message that is sent to the /broadcast-signal service endpoint.
+ */
+export interface BroadcastSignalBodyInterface {
+	/**
+	 * Content of signal. Required by server.
+	 */
+	signalContent: {
+		/**
+		 * Required parameter. Required by server. User may add more parameters than the required `type` and `content` below.
+		 */
+		contents: {
+			/**
+			 * Required parameter. User defined content. User defined content that will be pass
+			 * unchanged to client.
+			 */
+			content: JSON | string;
+			/**
+			 * Required parameter. User defined content that will be pass unchanged to client.
+			 * User may consider using this type to differentiate between multiple diffrent signal types,
+			 * and possibly use it for versioning as well.
+			 */
+			type: string;
+		};
+	};
+}
+
 /**
  * Submits notifications of changes to Fluid Service.
  */
 function echoExternalDataWebhookToFluid(
 	taskData: ITaskData,
 	fluidServiceUrl: string,
-	containerUrl: string,
 	externalTaskListId: string,
+	tenantId: string,
+	documentId: string,
 ): void {
+	const fluidService = `${fluidServiceUrl}/${tenantId}/${documentId}/broadcast-signal`;
 	console.log(
-		`CUSTOMER SERVICE: External data has been updated. Notifying Fluid Service at ${fluidServiceUrl}`,
+		`CUSTOMER SERVICE: External data has been updated. Notifying Fluid Service at ${fluidService}`,
 	);
 
-	// TODO: we will need to add details (like ContainerId) to the message body or the url,
-	// so this message body format will evolve
-	const messageBody = JSON.stringify({ taskData, containerUrl, externalTaskListId });
-	fetch(fluidServiceUrl, {
+	const messageBody = {
+		signalContent: {
+			contents: {
+				content: { taskData, externalTaskListId },
+				type: "ExternalDataChanged_V1.0.0",
+			},
+		},
+	} as unknown as BroadcastSignalBodyInterface;
+
+	fetch(fluidService, {
 		method: "POST",
 		headers: {
 			"Access-Control-Allow-Origin": "*",
 			"Content-Type": "application/json",
 		},
-		body: messageBody,
+		body: JSON.stringify(messageBody),
```

Comment:
Is stringifying necessary here? I know we made it so when `curl`ing the body could be a non-stringified JSON object, so I kind of expected the same here.

Reply:
It fails with this error if I don't stringify it. Oddly, when I console log it in tinylicious, it's a JSON - I think the content header does that translation at that point - even though I'm stringifying it here. 
<img width="977" alt="Screenshot 2023-10-18 at 8 20 27 AM" src="https://github.com/microsoft/FluidFramework/assets/6777404/1526d1e9-834b-4742-9d0a-5f0e599e87b4">